### ISSUE TITLE
buildNodeTree modifies original document

### ISSUE THREAD

**Issue by pspi** [lasar/dynalist-js#1]
Does buildNodeTree somehow modify the original document? 

This doesn't work:

```js
(async () => {
  const document = await dynalist.readDocument("lkmdfvlkmdfv");
  dynalist.util.buildNodeTree(document.nodes);
  dynalist.util.buildNodeMap(document.nodes);
})();
```

```
TypeError: Cannot set property 'parent' of undefined
    at node_modules/dynalist-js/src/utilities.js:32:46
    at Array.forEach (<anonymous>)
    at Utilities.buildNodeMap (node_modules/dynalist-js/src/utilities.js:26:24)
```
But when reversing the build lines, this works ok:

```js
(async () => {
  const document = await dynalist.readDocument("lkmdfvlkmdfv");
  dynalist.util.buildNodeMap(document.nodes);
  dynalist.util.buildNodeTree(document.nodes);
})();
```

**Comment by lasar** [comment]
Hi,

thanks for the bug report. You are right, the function inadvertedly modified the original document.

I've fixed the code, added some tests and published version 1.0.2 to NPM. Could you verify that the problem is gone now?

Take care,

_Lasar

**Comment by pspi** [comment]
Now it works. Thanks!

I'll close the issue.
